src/codecs/framers/length_delimited.rs

```diff
@@ -0,0 +1,162 @@
+use crate::codecs::{BoxedFramer, BoxedFramingError, FramingConfig};
+use bytes::{Bytes, BytesMut};
+use serde::{Deserialize, Serialize};
+use tokio_util::codec::Decoder;
+
+/// Config used to build a `LengthDelimitedCodec`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct LengthDelimitedDecoderConfig;
+
+#[typetag::serde(name = "length_delimited")]
+impl FramingConfig for LengthDelimitedDecoderConfig {
+    fn build(&self) -> crate::Result<BoxedFramer> {
+        Ok(Box::new(LengthDelimitedCodec::new()))
+    }
+}
+
+/// A codec for handling bytes sequences whose length is encoded in a frame head.
+///
+/// Currently, this expects a length header in 32-bit MSB by default; options to
+/// control the format of the header can be added in the future.
+#[derive(Debug)]
+pub struct LengthDelimitedCodec(tokio_util::codec::LengthDelimitedCodec);
+
+impl LengthDelimitedCodec {
+    /// Creates a new `LengthDelimitedCodec`.
+    pub fn new() -> Self {
+        Self(tokio_util::codec::LengthDelimitedCodec::new())
+    }
+}
+
+impl Default for LengthDelimitedCodec {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Clone for LengthDelimitedCodec {
+    fn clone(&self) -> Self {
+        // This is an awful implementation for `Clone` since it resets the
+        // internal state. However, it works for our use case because we
+        // generally only clone a codec that has not been mutated yet.
+        //
+        // Ideally, `tokio_util::codec::LengthDelimitedCodec` should implement
+        // `Clone` and it doesn't look like it was a deliberate decision to
+        // leave out the implementation. All of its internal fields implement
+        // `Clone`, so adding an implementation for `Clone` could be contributed
+        // to the upstream repo easily by adding it to the `derive` macro.
```

Comment:
Did we open an issue around this already?

Reply:
https://github.com/tokio-rs/tokio/pull/4089